compiler/test/dotty/tools/vulpix/VulpixUnitTests.scala

```diff
@@ -24,11 +24,9 @@ class VulpixUnitTests extends ParallelTesting {
   @Test def missingFile: Unit =
     try {
       compileFile("tests/vulpix-tests/unit/i-dont-exist.scala", defaultOptions).expectFailure.checkExpectedErrors()
-      fail("didn't fail properly")
-    }
-    catch {
+      throw new Exception("didn't fail properly")
```

Comment:
Maybe, factor this into a method `fail()` with a comment saying that JUnit `Assert.fail` cannot be used because it throws an `AssertionError` and that's what we expect to catch.